webapp/lib/data_model.dart

```diff
@@ -147,28 +148,36 @@ class Code {
     shortcut = codeMap.containsKey("Shortcut") ? codeMap["Shortcut"] : "";
     color = codeMap.containsKey("Color") ? codeMap["Color"] : "";
   }
+
+  toFirebaseMap() => {
+    "CodeID": id,
+    "DisplayText": displayText,
+    "NumericValue": numericValue,
+    "VisibleInCoda": visibleInCoda,
+    "Shortcut": shortcut, // Should this be optional?
+    "Color": color
+  };
 }
 
 class Origin {
   String id;
   String name;
   String originType;
-  Map<String, String> metadata;
+  Map<String, dynamic> metadata;
```

Comment:
I think I would prefer String : String, otherwise we could end up in a circumstance where a bug might be introduced that put some data in here that we couldn't serialise, which would crash the file save mechanism. Let's be restrictive for now and see if we come to regret it.